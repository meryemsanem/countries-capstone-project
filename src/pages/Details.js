import React from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { BiArrowBack } from 'react-icons/bi';
import './styles/Details.css';

function Details() {
  const { name } = useParams();
  const { data } = useSelector((state) => state.country);
  const countryDetails = data.find((country) => country.name === name);
  const navigation = useNavigate();

  if (!countryDetails) {
    return (
      <div className="not-found">
        <h2>Country Not Found</h2>
        <button
          type="button"
          onClick={() => navigation('/')}
          className="back-btn"
        >
          <BiArrowBack />
        </button>
      </div>
    );
  }

  return (
    <div className="details-container">
      <div className="details-country">
        <div className="background">
          <button
            type="button"
            onClick={() => navigation('/')}
            className="back-btn"
          >
            <BiArrowBack />
          </button>
        </div>
        <h2 className="header">
          {countryDetails.name}
          &apos;s Details
        </h2>
      </div>
      <div className="container">
        <img src={countryDetails.flag} alt={countryDetails.name} />
        <div className="flex-container">
          <h2 className="country-name-details">{countryDetails.name}</h2>
          <h3 className="country-details-off">
            Official Name:
            {' '}
            {countryDetails.officialName}
          </h3>
          <h4 className="country-details">
            Capital City:
            {' '}
            {countryDetails.capital}
          </h4>
          <h4 className="country-details">
            Continent:
            {' '}
            {countryDetails.continent}
          </h4>
          <h4 className="country-details">
            Population:
            {' '}
            {countryDetails.population}
          </h4>
          <h4 className="country-details">
            Area:
            {countryDetails.area}
          </h4>
          <h4 className="country-details">
            Timezone:
            {countryDetails.timezone[0]}
          </h4>
        </div>
      </div>
    </div>
  );
}

export default Details;
