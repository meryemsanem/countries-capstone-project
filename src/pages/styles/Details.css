.back-btn {
  color: white;
  border: none;
  background-color: #4468b0;
  font-size: 30px;
  cursor: pointer;
}

.container img {
  width: 30%;
  display: flex;
  margin: 40px auto;
}

.container {
  margin-inline: 10rem;
  gap: 10rem;
}

.country-name-details,
.country-details-off {
  text-align: center;
}

.flex-container {
  padding: 10px;
  background-color: #4468b0;
  border-radius: 40px;
  color: white;
  border: 1px solid #4468b0;
}

.details-country {
  background-color: #355087;
  height: 8rem;
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  flex-direction: column-reverse;
}

.header {
  text-align: center;
  color: white;
  align-items: center;
}

.details-container {
  background-color: #355087;
}

.not-found {
  text-align: center;
  color: #355087;
}
