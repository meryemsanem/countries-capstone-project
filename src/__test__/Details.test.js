import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Details from '../pages/Details';

const mockStore = configureMockStore([]);

const initialState = {
  country: {
    data: [
      {
        name: 'Test Country',
        flag: 'test-flag-url',
        officialName: 'Official Test Country',
        capital: 'Test Capital',
        continent: 'Test Continent',
        population: 'Test Population',
        area: 'Test Area',
        timezone: ['Test Timezone'],
      },
    ],
  },
};

describe('Details Component', () => {
  it('renders correctly', () => {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
