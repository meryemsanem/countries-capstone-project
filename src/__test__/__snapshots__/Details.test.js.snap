// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="not-found"
  >
    <h2>
      Country Not Found
    </h2>
    <button
      class="back-btn"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
